evm/hyperliquid-block-helper.js

```diff
@@ -0,0 +1,182 @@
+    const hash = actionHash(action, activePool, nonce);
+    const phantomAgent = constructPhantomAgent(hash, isMainnet);
+
+    const domain = {
+        chainId: 1337,
+        name: 'Exchange',
+        verifyingContract: '0x0000000000000000000000000000000000000000',
+        version: '1',
+    };
+
+    const types = {
+        Agent: [
+            { name: 'source', type: 'string' },
+            { name: 'connectionId', type: 'bytes32' },
+        ],
+        EIP712Domain: [
+            { name: 'name', type: 'string' },
+            { name: 'version', type: 'string' },
+            { name: 'chainId', type: 'uint256' },
+            { name: 'verifyingContract', type: 'address' },
+        ],
+    };
+
+    const signature = await wallet._signTypedData(domain, { Agent: types.Agent }, phantomAgent);
+    const { ethers } = require('ethers');
+    const sig = ethers.utils.splitSignature(signature);
+
+    return { r: sig.r, s: sig.s, v: sig.v };
+}
+
+async function sendRequest(action, signature, nonce, isMainnet) {
+    const payload = { action, signature, nonce };
+    const endpoint = isMainnet ? 'https://api.hyperliquid.xyz/exchange' : 'https://api.hyperliquid-testnet.xyz/exchange';
+
+    const curlCommand = `curl -s -X POST "${endpoint}" \
+        -H "Content-Type: application/json" \
+        -H "User-Agent: Mozilla/5.0 (compatible; Hyperliquid-Block-Helper/1.0)" \
+        -d '${JSON.stringify(payload)}' \
+        --connect-timeout 15 \
+        --max-time 30`;
+
+    const { stdout, stderr } = await execAsync(curlCommand);
+
+    if (stderr && !stderr.includes('curl')) {
+        throw new Error(`curl stderr: ${stderr}`);
+    }
+
+    const result = JSON.parse(stdout);
+    return result;
+}
+
+async function switchBlockSize(privateKey, useBig, network = 'mainnet') {
+    validateParameters({ isNonEmptyString: { privateKey } });
+
+    if (!privateKey.startsWith('0x')) {
+        privateKey = '0x' + privateKey;
+    }
+
+    if (privateKey.length !== 66) {
+        throw new Error(`Invalid private key length: ${privateKey.length}`);
+    }
+
+    const wallet = new Wallet(privateKey);
+    const isMainnet = network.toLowerCase() === 'mainnet';
+
+    const action = { type: 'evmUserModify', usingBigBlocks: useBig };
+    const nonce = Date.now();
+    const signature = await signL1Action(wallet, action, null, nonce, isMainnet);
+    const result = await sendRequest(action, signature, nonce, isMainnet);
+
+    if (result.status === 'ok') {
+        return result;
+    } else {
+        throw new Error(`API error: ${result.response}`);
+    }
+}
+
+async function processCommand(config, chain, options) {
+    const { privateKey, blockSize } = options;
+
+    validateParameters({
+        isNonEmptyString: { privateKey },
+        isNonEmptyString: { blockSize },
+    });
+
+    if (blockSize !== 'big' && blockSize !== 'small') {
+        throw new Error('Block size must be "big" or "small"');
+    }
```

Comment:
should be handled by CLI parsing definition. commander allows `.choices` to for options. see existing usage